### PR TITLE
CGMES export: export definition of regulating controls only once

### PR DESCRIPTION
**Pull request by zamarrenolm**
Signed-off-by: Luma <zamarrenolm@aia.es>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix

**What is the current behavior?** *(You can also link to an open issue here)*
If the same regulating control (RC) was referred by more than one equipment a separate definition for the RC was written to CGMES EQ output for each referral, resulting in multiple objects with same identifier being exported.

**What is the new behavior (if this is a feature change)?**
A regulating control definition is written only once in the CGMES EQ export, even if it is referred by multiple equipment.

### CODE REVIEW

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/RegulatingControlEq.java [hunk 1]
```diff
@@ -21,7 +24,17 @@ public final class RegulatingControlEq {
 
     public static final String REGULATING_CONTROL_VOLTAGE = "RegulatingControlModeKind.voltage";
 
-    public static void write(String id, String regulatingControlName, String terminalId, String cimNamespace, XMLStreamWriter writer) throws XMLStreamException {
+    public static String writeKindVoltage(Connectable<?> c, String terminalId, Set<String> regulatingControlsWritten, String cimNamespace, XMLStreamWriter writer) throws XMLStreamException {
```
Comment (miovd) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/RegulatingControlEq.java:27]: Just to be sure, why did you change the method's name?

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java [hunk 2]
```diff
@@ -433,10 +426,11 @@ private static <C extends Connectable<C>> void writePhaseTapChanger(C eq, PhaseT
             if (regulatingControlId.isPresent()) {
                 String mode = getPhaseTapChangerRegulationMode(ptc);
                 // Only export the regulating control if mode is valid
-                if (mode != null) {
+                if (mode != null && !regulatingControlsWritten.contains(regulatingControlId.get())) {
```
Comment (miovd) [inline cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java:429]: I think you should check before fetching `mode`:
```java
if (regulatingControlId.isPresent() && !regulatingControlsWritten.contains(regulatingControlId.get())
```

Comment (miovd) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java:429]: Yes, I meant to move the `get` method two lines before (like it is done below)

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/RegulatingControlEq.java (hunk 1 above)
Comment (miovd) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/elements/RegulatingControlEq.java:27]: Okay, thank you!

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java (hunk 2 above)
Comment (zamarrenolm) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/export/EquipmentExport.java:429]: Agree, better together. I was thinking of avoiding the check in the regulatingControlsWritten if mode was null, but your solutions is cleaner.

**Review by miovd — APPROVED** [review]
